osi_environment.proto

```diff
@@ -324,4 +336,148 @@ message EnvironmentalConditions
         //
         optional uint32 seconds_since_midnight = 1;
     }
+
+    //
+    // \brief Specification of the cloud states.
+    //
+    // \note within this message there is the cloud state level.
+    //
+    message Clouds
+    {
+        // Definition of the cloud state level, i.e. cloud state and sky visualization settings.
+        //
+        optional CloudStateLevel cloud_state_level = 1;
+
+        // Defines the cloud state levels given by observation of total cloud amount in oktas.
+        //
+        // The total degree of coverage indicates how large the part of the sky vault is which is covered
+        // with clouds altogether. It is given in eighths, because this division is easier to estimate
+        // for the observer. 0 eighths means that there are no traces of clouds in the sky,
+        // 4 eighths means that the sky is covered with clouds up to half, 8 eighths means that the sky is
+        // completely covered with clouds and no sky blue can be recognized.
+        //
+        // For cloud determination, the observer will choose a location from which he can overlook the entire sky.
+        // When determining the total cloud cover, all clouds or parts of clouds, regardless of height and genus,
+        // that are above his location will be added. Aircraft contrails are also counted as cloud cover if they remain
+        // in the sky for more than 15 minutes. On the other hand, fog is not counted as cloud cover.
+        //
+        // With a little practice, the observer can easily determine the total amount of cloud cover.
+        // Problems occur only with scattered clouds or many single clouds (cumulus).
+        // Their cloud cover is often overestimated. On the other hand,
+        // the coverage of thin ice clouds (cirrus) is often underestimated [3].
+        //
+        // \par References:
+        // [1] ISO 15469:2004(E) / CIE S 011/E:2003 : Spatial distribution of skylight
+        // [2] https://en.wikipedia.org/wiki/Cloud_cover
+        // [3] https://www.dwd.de/DE/service/lexikon/Functions/glossar.html?lv2=100932&lv3=101016
+        //
+        enum CloudStateLevel
+        {
+            // Cloud State is unknown (must not be used in ground truth).
+            //
+            CLOUD_STATE_LEVEL_UNKNOWN = 0;
+
+            // Other (unspecified but known) CloudState.
+            //
+            CLOUD_STATE_LEVEL_OTHER = 1;
+
+            // Cloudless (0/8)
+            //
+            CLOUD_STATE_LEVEL_CLOUDLESS = 2;
+
+            // Sunny (1/8)
+            //
+            CLOUD_STATE_LEVEL_SUNNY = 3;
```

Comment:
Please update the Enums to better align with commonly used terminologies.
For example "CLOUD_STATE_LEVEL_SUNNY " is miss leading at night.
For reference: https://www.eoas.ubc.ca/courses/atsc113/flying/met_concepts/01-met_concepts/01c-cloud_coverage/index.html

Reply:
Tank you for your feedback. I could change the SUNNY level to `CLOUD_STATE_LEVEL_CLEAR` or I could change all the states to `CLOUD_STATE_LEVEL_1 - 8`?
What do you think about those ideas?

Reply:
After a quick research it seems that there are not standardized (international) names for the octas available: according to https://www.eoas.ubc.ca/courses/atsc113/flying/met_concepts/01-met_concepts/01c-cloud_coverage/index.html 
different octas are combined to one shared name.
In German there seems to be a distinct naming for the octas, but also not ideal since it uses "sunny" for the cloud coverage which does not fit at night.

I would suggest not to invent new octa names. Instead I would name it to CLOUD_STATE_LEVEL_0..8 and give the reference and the names according to https://www.eoas.ubc.ca/courses/atsc113/flying/met_concepts/01-met_concepts/01c-cloud_coverage/index.html (or: From Stull, 2017: Practical Meteorology: An Algebra-based Survey of Atmospheric Science.) in the comments.
@pmai What do you prefer?

Reply:
I talked to Pierre about what we discussed in the previous Harmonisation-Meeting: Naming the enums using 0-8 (Oktas) is probably the best idea. Also, we should definitely add a "Sky Obscured" enum since "Other" usually refers to something completely different (not supported by OSI) but known. We **could** also add "Not Measured" (as described here: https://www.eoas.ubc.ca/courses/atsc113/flying/met_concepts/01-met_concepts/01c-cloud_coverage/index.html) as separate enum. Though, this would not make sense in Ground Truth anyway and for Sensor output side there is no measured/detected Environmental Condition in OSI.